tests/routes/test_subject.py

```diff
@@ -0,0 +1,79 @@
+import pytest
+from httpx import AsyncClient
+from fastapi import status
+from json import loads
+
+from tests.utils.subject_test_utils import add_subject, make_subject_register_input
+from tests.utils.default_values.test_subject_default_values import SubjectDefaultValues
+
+from server.models.database.subject_db_model import Subject
+from server.models.database.user_db_model import User
+
+MAX_SUBJECT_COUNT = 5
+UPDATED_SUBJECT_CODE = "DEF000U"
+
+
+@pytest.mark.asyncio
+async def test_subject_get_all(client: AsyncClient, user: User) -> None:
```

Comment:
Esse teste precisa executar a fixture de criar usuário admin, mas não usar a entidade de usuário. Para esse caso, o melhor a se fazer não é requerir o objeto do tipo usuário, mas sim marcar o teste para usar a fixture com pytest.mark.usefixtures()

https://docs.pytest.org/en/7.1.x/how-to/fixtures.html#use-fixtures-in-classes-and-modules-with-usefixtures